cmd/kube-apiserver/app/server.go

```diff
@@ -101,7 +101,11 @@ cluster's shared state through which all other components interact.`,
 
 // Run runs the specified APIServer.  This should never exit.
 func Run(runOptions *options.ServerRunOptions, stopCh <-chan struct{}) error {
-	kubeAPIServerConfig, sharedInformers, insecureServingOptions, err := CreateKubeAPIServerConfig(runOptions)
+	tunneler, proxyTransport, err := CreateDialer(runOptions)
```

Comment:
I think the next step for this code is to figure out how to package up this transport and the service resolver into a single thing that can be passed around. That can happen later though.